charts/k8s-monitoring/templates/agent_config/_annotation_autodiscovery.river.txt

```diff
@@ -0,0 +1,79 @@
+{{ define "agent.config.annotationAutodiscovery" }}
+// Annotation Autodiscovery
+discovery.relabel "annotation_autodiscovery" {
+  targets = concat(discovery.kubernetes.pods.targets,discovery.kubernetes.services.targets)
+  rule {
+    source_labels = [
+      "__meta_kubernetes_pod_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.scrape }}",
+      "__meta_kubernetes_service_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.scrape }}",
+    ]
+    regex = ";?true;?"
+    action = "keep"
+  }
+  rule {
+    source_labels = [
+      "__meta_kubernetes_pod_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.job }}",
+      "__meta_kubernetes_service_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.job }}",
+    ]
+    regex = ";?([^;]+);?"
+    action = "replace"
+    target_label = "job"
+  }
+  rule {
+    source_labels = [
+      "__meta_kubernetes_pod_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.instance }}",
+      "__meta_kubernetes_service_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.instance }}",
+    ]
+    regex = ";?([^;]+);?"
+    action = "replace"
+    target_label = "instance"
+  }
+  rule {
+    source_labels = [
+      "__meta_kubernetes_pod_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.metricsPath }}",
+      "__meta_kubernetes_service_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.metricsPath }}",
+    ]
+    regex = ";?([^;]+);?"
+    action = "replace"
+    target_label = "__metrics_path__"
+  }
+  rule {
+    source_labels = [
+      "__address__",
+      "__meta_kubernetes_pod_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.metricsPort }}",
+      "__meta_kubernetes_service_annotation_{{ include "escape_label" .Values.metrics.autoDiscover.annotations.metricsPort }}",
+    ]
+    regex = "(.+);;?([0-9]+);?"
```

Comment:
Is the double `;;` intentional?

Reply:
Yeah, because we're trying to handle either the case of pods or services. If a pod matches, then the value will be `<address>;8080;` and if a service matches, it'll be `<address>;;8080`. The regex matches the address and the port, no matter if the port was set from the pod annotation or the service annotation.

Reply:
FWIW, after chatting with Craig, I split that discovery.relabel component into two, one for services and one for pods. That'll clean things up a bunch.